### DISCUSSION TITLE
Image preview link

### DISCUSSION THREAD

**Discussion by seghier** [Comfy-Org/ComfyUI#3522] [Q&A]
Hello
Is there a way to get the link of preview image or a temp file path in realtime? 
I see that the image link is like this
`blob:http://127.0.0.1:8188/c1058cde-74aa-4852-93f6-7b9fb25df03d`
and it change every time

**Comment by ltdrdata** [comment]
The preview that appears during execution is not stored anywhere and is only temporarily valid while being transmitted via websocket.

If you want the results of the PreviewImage or SaveImage nodes, you have to use url like this.
`http://127.0.0.1:8188/view?filename=ComfyUI_temp_kdprd_00001_.png&type=input&subfolder=`
type can be input/output/temp

Use temp type if you want to access PreviewImage's result.
Use output type if you want to access SaveImage's result.

**Comment by seghier** [comment]
Thank you @ltdrdata , looks like i stay with the classic method to get the latest images from output folder

**Comment by julienduranton** [comment]
I've done it by checking the type of the event in the eventListener (try catch on the parsing of event.data).
The previews will be in that blob that I transform like such:

`  socket.addEventListener('message', async (event) => {
    try {
      data = JSON.parse(event.data);
    }
    catch {
      // console.log(event);
      await blobToBase64(event.data)
    } `

`  const blobToBase64 = async blob => {
    let result_base64 = await new Promise((resolve) => {
      let fileReader = new FileReader();
      fileReader.onload = (e) => resolve(fileReader.result);
      fileReader.readAsArrayBuffer(blob);
    });
    var res = new Uint8Array(result_base64, 8);
    var new_blob = new Blob([res], { type: 'image/jpeg' });
    var new_url =  URL.createObjectURL( new_blob);
    myImage.src = new_url
  }`

**Reply by julienduranton** [reply]
sorry about the disgusting format ...
